sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/PCollectionConsumerRegistry.java

```diff
@@ -210,8 +233,8 @@ public void accept(WindowedValue<T> input) throws Exception {
       try (Closeable close =
           MetricsEnvironment.scopedMetricsContainer(this.unboundMetricContainer)) {
         // Increment the counter for each window the element occurs in.
-        this.counter.inc(input.getWindows().size());
-
+        this.elementCountCounter.inc(input.getWindows().size());
+        this.sampledByteSizeDistribution.tryUpdate(input.getValue(), this.coder);
```

Comment:
Should the size also be multiplied by # of window even though for most cases the number of window is always 1?

Reply:
Since this is a mean value, don't we want to keep it the metric size of 1 window?

Reply:
It's not only a mean value but a distribution value(min, max, sum count), right? Can we put a TODO here to take care the case when we have multiple windows optimization some day? Besides, I was wrong in my comment. When considering multiple windows, the size should not be multiplied but the counter should be updated  per window with the same element size.